src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -83,41 +126,51 @@ private static JObject ReadProject(string projectJsonPath)
            } 
         }
 
+        private JToken GetFrameworkDependenciesSection(JObject projectJsonRoot, string framework = null)
+        {
+            if(string.IsNullOrWhiteSpace(framework))
```

Comment:
Perhaps I'm still missing something you are describing.  The way it works, in this code, is that if you do not specify a framework, then we grab all of the frameworks sections, and add the dependencies to each of those...

            if (Frameworks == null || Frameworks.Length == 0)
            {
                Frameworks = projectRoot.SelectTokens("frameworks").SelectMany(f => f.Children().Select(c => ((JProperty)c).Name)).ToArray();
            }
            for (int i = 0; i < Frameworks.Length; i++)
            {
                JObject dependencies = GenerateDependencies(projectRoot, Frameworks[i]);
                projectRoot = UpdateProperty(projectRoot, dependencies, Frameworks[i]);
            }

The method you are referencing is called from GenerateDependencies, so we update each framework with the new dependencies.